torch/csrc/jit/graph_executor.cpp

```diff
@@ -331,17 +330,13 @@ struct GraphExecutorImpl {
   }
 
   GraphExecutorImpl(std::shared_ptr<Graph> graph, bool optimize)
-    : graph(prepareGraph(graph))
-    , optimize(optimize)
-    , num_inputs(this->graph->inputs().size())
-    , num_flat_inputs(countFlatInputs(graph))
-    , num_outputs(this->graph->outputs().size()) {
-      // until we have correct alias analysis any use of mutable operators
-      // disables all optimization
-      if(hasMutableOperators(this->graph->block())) {
-        optimize = false;
-      }
-    }
+      : graph(prepareGraph(graph)),
+        // until we have correct alias analysis any use of mutable operators
+        // disables all optimization
+        optimize(!hasMutableOperators(this->graph->block())),
```

Comment:
btw, here's an example of where we should have followed the FB style guide; `optimize` was getting shadowed in the constructor and so we weren't actually setting it. Using `optimize_` to denote the member variable status would have disambiguated

Reply:
good catch! The current code you have here ignores the `optimize` parameter, though.